quad_pyblish_module/plugins/tvpaint/publish/extract_psd.py

```diff
@@ -55,7 +55,7 @@ def process(self, instance):
                 files = repre['files']
 
             new_filenames = []
-            for filename in files:
+            for offset, filename in enumerate(files):
                 new_filename = os.path.splitext(filename)[0]
                 dst_filepath = os.path.join(repre["stagingDir"], new_filename)
```

Comment:
And for this one `Path(repre["stagingDir"], new_filename)`

Reply:
And it's weird to not use the `output_dir = instance.data.get("stagingDir")` created and checked on line `45`

Reply:
And the `output_dir` retrival or creation code is done on the for loop but it's something generic that should be done before the loop or this should be edited to use `repre["stagingDir"]` as value not `instance.data.get("stagingDir")`.

Maybe using `instance.data.get("stagingDir")` only has fallback value if exists.

Reply:
In fact it is not really unused, it creates the folder if it doesn't exists. But it won't work if the retrieved value is `None`, because the folder would not be setted to the correct attributes if created at this moment.
I'll update this portion of code to make it more logic and functionnal.